src/frontend/js/widgets/SyllabusCourseRunsList/components/CourseProductItem/index.spec.tsx

```diff
@@ -586,6 +586,65 @@ describe('CourseProductItem', () => {
     expect(screen.queryByTestId('PurchaseButton__cta')).toBeNull();
   });
 
+  it('renders a button and a banner if the contract needs to be signed', async () => {
+    const relation = CourseProductRelationFactory().one();
+    const { product } = relation;
+    const order = CredentialOrderFactory({
+      product: product.id,
+      target_courses: product.target_courses,
+      course: CourseLightFactory({ code: '00000' }).one(),
+      state: OrderState.SUBMITTED,
+    }).one();
+    fetchMock.get(`https://joanie.test/api/v1.0/courses/00000/products/${product.id}/`, relation);
+    const orderQueryParameters = {
+      product: order.product,
+      course: order.course.code,
+      state: ACTIVE_ORDER_STATES,
+    };
+
+    fetchMock.get(
+      `https://joanie.test/api/v1.0/orders/?${queryString.stringify(orderQueryParameters)}`,
+      [order],
+    );
+
+    render(
+      <Wrapper withSession>
+        <CourseProductItem productId={product.id} courseCode="00000" />
+      </Wrapper>,
+    );
+
+    // Wait for product information to be fetched
+    const loadingMessage = screen.getByRole('status', { name: 'Loading product information...' });
+    await waitForElementToBeRemoved(loadingMessage);
+    await screen.findByRole('heading', { level: 3, name: product.title });
+
+    // - A banner should be displayed.
+    screen.getByText('You need to sign your contract before enrolling to course runs');
+
+    screen.getByRole('link', { name: 'Go to signature' });
+
+    // - In place of product price, a label "Pending" should be displayed
+    const $enrolledInfo = await screen.findByText('Pending');
+    expect($enrolledInfo.tagName).toBe('STRONG');
+    expect($enrolledInfo.classList.contains('h6')).toBe(true);
```

Comment:
do not test the design.

Reply:
This assertion was already present in most tests of this file previously